File: src/j/implementation/InnerSet_1.java
package j.implementation;

public class InnerSet_1<T> implements InnerSet<T>
{	
	private T element_0;
	
	public InnerSet_1()
	{
		
	}
	
	public InnerSet_1(T p_element)
	{
		element_0 = p_element;
	}

	@Override
	@SuppressWarnings({ "rawtypes", "unchecked" })
	public InnerSet<T>addElement(T p_element)
	{
		if(element_0.equals(p_element))
			return this;
		else
			return new InnerSet_2(element_0, p_element);
	}

	@Override
	@SuppressWarnings({ "rawtypes", "unchecked" })
	public InnerSet<T> removeElement(T p_element)
	{
		if(element_0.equals(p_element))
				return new InnerSet_0();
		else
			return this;
	}
	
	@Override
	public String toString()
	{
		StringBuilder string_builder = new StringBuilder();
		
		string_builder.append("{");
		
		string_builder.append(element_0);
		
		string_builder.append("}");
		
		return string_builder.toString();
	}
}

File: src/csharp/InnerSet_1.cs
using System;
namespace OSSL
{
	public class InnerSet_1<T> : InnerSet<T>
	{
		public T a_element_1
		{
			get
			{
				return a_element_1;
			}
			set
			{
				a_element_1 = value;
			}
		}

		public InnerSet_1()
		{
		}

		public InnerSet_1(T p_element)
		{
			a_element_1 = p_element;
		}

		public InnerSet<T> addElement(T p_element)
		{
			if (p_element.Equals(a_element_1))
				return new InnerSet_2<T>(this, p_element);
			else
				return this;
		}

		public InnerSet<T> removeElement(T p_element)
		{
			if (a_element_1.Equals(p_element))
				return new InnerSet_0<T>();
			else
				return this;
		}
	}
}
